pyogrio/tests/test_geopandas_io.py

```diff
@@ -411,58 +411,94 @@ def test_read_sql_dialect_sqlite_gpkg(naturalearth_lowres):
 
 
 @pytest.mark.parametrize(
-    "driver,ext",
-    [
-        ("ESRI Shapefile", "shp"),
-        ("GeoJSON", "geojson"),
-        ("GeoJSONSeq", "geojsons"),
-        ("GPKG", "gpkg"),
-    ],
+    "ext",
+    ALL_EXTS,
 )
-def test_write_dataframe(tmpdir, naturalearth_lowres, driver, ext):
-    expected = read_dataframe(naturalearth_lowres)
-
-    filename = os.path.join(str(tmpdir), f"test.{ext}")
-    write_dataframe(expected, filename, driver=driver)
+def test_write_dataframe(tmp_path, naturalearth_lowres, ext):
+    input_gdf = read_dataframe(naturalearth_lowres)
+    output_path = tmp_path / f"test{ext}"
 
-    assert os.path.exists(filename)
+    if ext == ".fgb":
+        # For .fgb, spatial_index=False to evade the rows being reordered
+        write_dataframe(input_gdf, output_path, spatial_index=False)
+    else:
+        write_dataframe(input_gdf, output_path)
 
-    df = read_dataframe(filename)
+    assert output_path.exists()
+    result_gdf = read_dataframe(output_path)
 
-    if driver != "GeoJSONSeq":
-        # GeoJSONSeq driver I/O reorders features and / or vertices, and does
-        # not support roundtrip comparison
+    geometry_types = result_gdf.geometry.type.unique()
+    if DRIVERS[ext] in DRIVERS_NO_MIXED_SINGLE_MULTI:
+        assert geometry_types == ["MultiPolygon"]
+    else:
+        assert len(set(geometry_types).intersection(("Polygon", "MultiPolygon"))) == 2
+
+    # Coordinates are not precisely equal when written to JSON
+    # dtypes do not necessarily round-trip precisely through JSON
+    is_json = ext in [".json", ".geojson", ".geojsonl"]
+    # In .geojsonl the vertices are reordered, so normalize
+    is_jsons = ext == ".geojsonl"
+
+    assert_geodataframe_equal(
+        result_gdf,
+        input_gdf,
+        check_less_precise=is_json,
+        check_index_type=False,
+        check_dtype=not is_json,
+        normalize=is_jsons,
+    )
 
-        # Coordinates are not precisely equal when written to JSON
-        # dtypes do not necessarily round-trip precisely through JSON
-        is_json = driver == "GeoJSON"
 
-        assert_geodataframe_equal(
-            df,
-            expected,
-            check_less_precise=is_json,
-            check_index_type=False,
-            check_dtype=not is_json,
+def to_multipolygon(geometries):
```

Comment:
Is this still used?

Reply:
Yes. There are 2 issues:
- when writing to json int fields are saved as int32 instead of int64 (or the other way around), which accounts for the ``check_dtype= not json``. I think you recently made an issue for this or something similar.
- when writing to json the coordinates are by default saved with 15 figures after the decimal seperator, and the rest is dropped. Using a layer creation option it is possible to specify a different number. I just tried it, and eg. with 20 figures the check_less_precise isn't needed anymore for this dataset.

Reply:
> * when writing to json int fields are saved as int32 instead of int64 (or the other way around), which accounts for the `check_dtype= not json`. I think you recently made an issue for this or something similar.

Indeed -> https://github.com/geopandas/pyogrio/issues/90

My comment was actually about the `to_multipolygon` function that is added, and not about the use of `check_dtype/check_less_precise=is_json` (that looks OK)

Reply:
Ah, indeed, the ``to_multipolygon`` function is indeed obsolete!